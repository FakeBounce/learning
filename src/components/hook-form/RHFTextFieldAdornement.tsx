import Iconify from '@src/components/iconify/Iconify';
import { memo, ReactNode } from 'react';
import { useFormContext, Controller } from 'react-hook-form';
import { IconButton, InputAdornment, TextField } from '@mui/material';

interface RHFTextFieldProps {
  name: string;
  helperText?: ReactNode;
  InputProps?: Record<string, any>;
  onClick: () => void;
  icon: string;
  [x: string]: any;
}

function RHFTextFieldAdornement({
  name,
  helperText,
  InputProps,
  onClick,
  icon,
  ...other
}: RHFTextFieldProps) {
  const { control } = useFormContext();

  return (
    <Controller
      name={name}
      control={control}
      render={({ field, fieldState: { error } }) => (
        <TextField
          {...field}
          fullWidth
          value={typeof field.value === 'number' && field.value === 0 ? '' : field.value}
          error={!!error}
          helperText={error ? error?.message : helperText}
          InputProps={{
            endAdornment: (
              <InputAdornment position="end">
                <IconButton onClick={onClick} edge="end">
                  <Iconify icon={icon} />
                </IconButton>
              </InputAdornment>
            ),
            ...InputProps
          }}
          {...other}
        />
      )}
    />
  );
}

export default memo(RHFTextFieldAdornement);
