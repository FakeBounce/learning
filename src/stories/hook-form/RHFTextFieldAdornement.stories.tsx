import { Meta, StoryObj } from '@storybook/react';
import { FormProvider, useForm } from 'react-hook-form';
import RHFTextFieldAdornement from '@src/components/hook-form/RHFTextFieldAdornement';
import { useState } from 'react';

const meta: Meta = {
  title: 'Example/RHFTextFieldAdornement',
  component: RHFTextFieldAdornement,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  }
};

export default meta;
type Story = StoryObj<typeof RHFTextFieldAdornement>;

const RHFTextFieldAdornementTemplate: Story = {
  render: (args: any) => {
    const [isPasswordVisible, setIsPasswordVisible] = useState(false);
    const methods = useForm();

    return (
      <FormProvider {...methods}>
        <form>
          <RHFTextFieldAdornement
            type={isPasswordVisible ? 'text' : 'password'}
            icon={isPasswordVisible ? 'eva:eye-fill' : 'eva:eye-off-fill'}
            onClick={() => setIsPasswordVisible(!isPasswordVisible)}
            {...args}
          />
        </form>
      </FormProvider>
    );
  }
};

export const Default = {
  ...RHFTextFieldAdornementTemplate,
  args: {
    name: 'Default',
    label: 'Default Field',
    defaultValue: ''
  }
};

export const WithPasswordShown = {
  ...RHFTextFieldAdornementTemplate,
  args: {
    name: 'pass',
    type: 'text',
    icon: 'eva:eye-fill',
    label: 'With password shown',
    defaultValue: 'qkdoaznd'
  }
};

export const WithPasswordHidden = {
  ...RHFTextFieldAdornementTemplate,
  args: {
    name: 'pass',
    label: 'With password hidden',
    defaultValue: 'qkdoaznd',
    icon: 'eva:eye-off-fill'
  }
};
