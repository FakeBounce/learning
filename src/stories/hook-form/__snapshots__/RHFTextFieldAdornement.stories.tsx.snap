// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/RHFTextFieldAdornement Default smoke-test 1`] = `
<form>
  <div class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root">
    <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-ioiz93-MuiFormLabel-root-MuiInputLabel-root"
           data-shrink="false"
           for=":r0:"
           id=":r0:-label"
    >
      Default Field
    </label>
    <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-fn92bf-MuiInputBase-root-MuiOutlinedInput-root">
      <input aria-invalid="false"
             id=":r0:"
             name="Default"
             type="password"
             class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-t0xxjf-MuiInputBase-input-MuiOutlinedInput-input"
             value
      >
      <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-93rqj-MuiInputAdornment-root">
        <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ql7g3c-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
        >
          <span>
          </span>
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
      </div>
      <fieldset aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend class="css-yjsfm1">
          <span>
            Default Field
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</form>
`;

exports[`Example/RHFTextFieldAdornement WithPasswordHidden smoke-test 1`] = `
<form>
  <div class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root">
    <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-3pz6tx-MuiFormLabel-root-MuiInputLabel-root"
           data-shrink="true"
           for=":r2:"
           id=":r2:-label"
    >
      With password hidden
    </label>
    <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-fn92bf-MuiInputBase-root-MuiOutlinedInput-root">
      <input aria-invalid="false"
             id=":r2:"
             name="pass"
             type="password"
             class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-t0xxjf-MuiInputBase-input-MuiOutlinedInput-input"
             value="qkdoaznd"
      >
      <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-93rqj-MuiInputAdornment-root">
        <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ql7g3c-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
        >
          <svg xmlns="http://www.w3.org/2000/svg"
               xmlns:xlink="http://www.w3.org/1999/xlink"
               aria-hidden="true"
               role="img"
               class="MuiBox-root css-1t9pz9x iconify iconify--eva"
               width="1em"
               height="1em"
               viewbox="0 0 24 24"
          >
            <circle cx="12"
                    cy="12"
                    r="1.5"
                    fill="currentColor"
            >
            </circle>
            <path fill="currentColor"
                  d="M15.29 18.12L14 16.78l-.07-.07l-1.27-1.27a4.07 4.07 0 0 1-.61.06A3.5 3.5 0 0 1 8.5 12a4.07 4.07 0 0 1 .06-.61l-2-2L5 7.87a15.89 15.89 0 0 0-2.87 3.63a1 1 0 0 0 0 1c.63 1.09 4 6.5 9.89 6.5h.25a9.48 9.48 0 0 0 3.23-.67ZM8.59 5.76l2.8 2.8A4.07 4.07 0 0 1 12 8.5a3.5 3.5 0 0 1 3.5 3.5a4.07 4.07 0 0 1-.06.61l2.68 2.68l.84.84a15.89 15.89 0 0 0 2.91-3.63a1 1 0 0 0 0-1c-.64-1.11-4.16-6.68-10.14-6.5a9.48 9.48 0 0 0-3.23.67Zm12.12 13.53L19.41 18l-2-2l-9.52-9.53L6.42 5L4.71 3.29a1 1 0 0 0-1.42 1.42L5.53 7l1.75 1.7l7.31 7.3l.07.07L16 17.41l.59.59l2.7 2.71a1 1 0 0 0 1.42 0a1 1 0 0 0 0-1.42"
            >
            </path>
          </svg>
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
      </div>
      <fieldset aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend class="css-14lo706">
          <span>
            With password hidden
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</form>
`;

exports[`Example/RHFTextFieldAdornement WithPasswordShown smoke-test 1`] = `
<form>
  <div class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root">
    <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-3pz6tx-MuiFormLabel-root-MuiInputLabel-root"
           data-shrink="true"
           for=":r1:"
           id=":r1:-label"
    >
      With password shown
    </label>
    <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-fn92bf-MuiInputBase-root-MuiOutlinedInput-root">
      <input aria-invalid="false"
             id=":r1:"
             name="pass"
             type="text"
             class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-t0xxjf-MuiInputBase-input-MuiOutlinedInput-input"
             value="qkdoaznd"
      >
      <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-93rqj-MuiInputAdornment-root">
        <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ql7g3c-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
        >
          <span>
          </span>
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
      </div>
      <fieldset aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend class="css-14lo706">
          <span>
            With password shown
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</form>
`;
